src/components/dialogs/free-properties-based-filter.js

```diff
@@ -0,0 +1,577 @@
+
+    const predefinedValues = useMemo(() => {
+        const predefinedForName = predefined?.[name];
+        if (!predefinedForName) return [];
+        return [...new Set(predefinedForName)].sort();
+    }, [name, predefined]);
+
+    const [values, setValues] = useState(defaultValue?.values || []);
+
+    useEffect(() => {
+        setName(defaultValue?.name);
+        setValues(defaultValue?.values || []);
+    }, [defaultValue]);
+
+    return (
+        <>
+            <FormControl
+                fullWidth
+                margin="dense"
+                sx={{ flexBasis: '28%', marginTop: 0 }}
+            >
+                <Autocomplete
+                    id={'name_property'}
+                    defaultValue={''}
+                    value={name}
+                    freeSolo
+                    autoSelect
+                    forcePopupIcon
+                    onChange={(oldVal, newVal) => {
+                        onChange(index, { name: newVal, values: [] });
+                        setName(newVal);
+                        setValues([]);
+                    }}
+                    options={predefinedNames}
+                    renderInput={(props) => (
+                        <TextField
+                            label={<FormattedMessage id="PropertyName" />}
+                            error={!!errors?.PropName}
+                            {...props}
+                        />
+                    )}
+                />
+            </FormControl>
+            <FormControl
+                sx={{ paddingLeft: '1ex', flexBasis: '72%', marginTop: 0 }}
+            >
+                <Autocomplete
+                    id="prop_values"
+                    value={values}
+                    freeSolo
+                    autoSelect
+                    forcePopupIcon
+                    multiple={true}
+                    onChange={(oldVal, newVal) => {
+                        onChange(index, { name, values: newVal });
+                        setValues(newVal);
+                    }}
+                    options={predefinedValues}
+                    renderInput={(props) => (
+                        <TextField
+                            label={<FormattedMessage id="PropertyValues" />}
+                            error={!!errors?.PropValue}
+                            {...props}
+                        />
+                    )}
+                    renderTags={(val, getTagsProps) =>
+                        val.map((code, index) => (
+                            <Chip
+                                id={'chip_' + code}
+                                size={'small'}
+                                label={code}
+                                {...getTagsProps({ index })}
+                            />
+                        ))
+                    }
+                />
+            </FormControl>
+        </>
+    );
+};
+
+const makeFreePropertiesProblemsMap = (values, full, prevErrors) => {
```

Comment:
this function is clone of makeFreeProperties2ProblemsMap, except one line.
could we use a single function .